File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heap=[]
    
    answer = 0
    
    for i in range(len(scoville)):
        heapq.heappush(heap,scoville[i])
    
    flag = False
    for i in range(len(scoville)):
        
        x1 = heapq.heappop(heap)
        
        if K <=x1:
            flag =True
            break
        if len(scoville)-1== i:
            break
        else:
            x2 = heapq.heappop(heap)
            heapq.heappush(heap,x1+(x2*2))
            
            answer +=1
    if flag == False:
        answer = -1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        boolean flag = false;
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        
        for (int i = 0; i<scoville.length; i++){
            heap.add(scoville[i]);
        }
        
        
        for (int i = 0; i< scoville.length; i++){
            int x1 = heap.poll();
            
            if (x1 >=K){
                flag = true;
                break;
            }
            if (scoville.length -1 == i){
                break;
            }
            else{
                int x2 = heap.poll();
                heap.add(x1+(x2*2));
                answer += 1;
            }
        }
        if(flag == false){
            answer = -1;
        }
        
        return answer;
    }
}